src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +61,646 @@ const ActorInit En_Rz_InitVars = {
+    EnRz_UpdateSkelAnime(this, play);
+
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        this->actionFunc = EnRz_Walk;
+        func_80BFB9E4(play, this, EN_RZ_ANIM_WALKING);
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC728;
+        func_80BFB9E4(play, this, EN_RZ_ANIM_THINKING);
+        this->actor.speedXZ = 0.0f;
+        func_80BFBDFC(play);
+    } else if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void EnRz_StopToThink(EnRz* this, PlayState* play) {
+    this->timer = 100;
+    this->actionFunc = func_80BFC7E0;
+    this->actor.speedXZ = 0.0f;
+    func_80BFB9E4(play, this, EN_RZ_ANIM_THINKING);
+}
+
+void EnRz_Walk(EnRz* this, PlayState* play) {
+    EnRz_UpdateSkelAnime(this, play);
+    this->actor.speedXZ = 1.5f;
+
+    switch (EnRz_PathStatus(this)) {
+        case 2:
+            EnRz_SetupPath(this, play);
+            EnRz_StopToThink(this, play);
+            break;
+
+        case 1:
+            EnRz_StopToThink(this, play);
+            break;
+
+        default:
+            break;
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC728;
+        func_80BFB9E4(play, this, EN_RZ_ANIM_THINKING);
+        this->actor.speedXZ = 0.0f;
+        func_80BFBDFC(play);
+    } else if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void EnRz_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRz* this = THIS;
+
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+
+    this->actionFunc(this, play);
+
+    if (DECR(this->blinkTimer) == 0) {
+        this->blinkTimer = Rand_S16Offset(60, 60);
+    }
+
+    this->eyeIndex = this->blinkTimer;
+    if (this->eyeIndex > 2) {
+        this->eyeIndex = 0;
+    }
+}
+
+void EnRz_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnRz* this = THIS;
+    static Vec3f focusOffsetPos = { 500.0f, -500.0f, 0.0f };
+
+    if (limbIndex == OBJECT_RZ_LIMB_0B) {
+        Matrix_MultVec3f(&focusOffsetPos, &thisx->focus.pos);
+    }
+    if (limbIndex == OBJECT_RZ_LIMB_03) {
+        Matrix_MultVec3f(&gZeroVec3f, &this->shadowPos);
+    }
+}
+
+void EnRz_Draw(Actor* thisx, PlayState* play) {
+    EnRz* this = THIS;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    func_8012C28C(play->state.gfxCtx);
+
+    if (EN_RZ_GET_SISTER(&this->actor) == EN_RZ_JUDO) {
+        AnimatedMat_DrawStepOpa(play, Lib_SegmentedToVirtual(&object_rz_Matanimheader_00D768), EN_RZ_JUDO);
+    } else { // EN_RZ_MARILLA
+        AnimatedMat_DrawStepOpa(play, Lib_SegmentedToVirtual(&object_rz_Matanimheader_00D768), EN_RZ_MARILLA);
```

Comment:
The enum use doesn't seem appropriate here. The last arg to `AnimatedMat_DrawStepOpa` is step, not a sister type.

Reply:
![image](https://user-images.githubusercontent.com/96613413/195725036-e449b812-2769-42db-b336-c6cb812906bf.png)

Reply:
@EllipticEllipsis were you referring to the last argument as well or just the `if else`? Again the last argument using that enum just doesn't really make sense to me.

Reply:
I think you're right, he was just referring to the if else only